tools/src/bin/collect.rs

```diff
@@ -389,115 +466,208 @@ async fn main() -> anyhow::Result<()> {
-                    .map_err(|err| Error::Anyhow(err.into()))?;
-            run_collection_generic(options, vdaf, http_client, query, &()).await
-        }
-        #[cfg(feature = "fpvec_bounded_l2")]
-        (VdafType::FixedPoint32BitBoundedL2VecSum, Some(length), None) => {
-            let vdaf: Prio3FixedPointBoundedL2VecSumMultithreaded<FixedI32<U31>> =
-                Prio3::new_fixedpoint_boundedl2_vec_sum_multithreaded(2, length)
-                    .map_err(|err| Error::Anyhow(err.into()))?;
-            run_collection_generic(options, vdaf, http_client, query, &()).await
+    options_dispatch!(options, (query, vdaf) => {
+        match options.subcommand {
+            Some(Subcommands::NewJob { collection_job_id }) => {
+                let collection_job_id = collection_job_id.unwrap_or_else(random);
+                run_new_job(options, vdaf, http_client, query, &(), collection_job_id).await
+            }
+            Some(Subcommands::PollJob { collection_job_id }) => {
+                run_poll_job(options, vdaf, http_client, query, (), collection_job_id).await
+            }
+            _ => run_collection(options, vdaf, http_client, query, &()).await,
         }
-        _ => Err(clap::Error::raw(
-            ErrorKind::ArgumentConflict,
-            format!(
-                "incorrect VDAF parameter arguments were supplied for {}",
-                options
-                    .vdaf
-                    .to_possible_value()
-                    .unwrap()
-                    .get_help()
-                    .unwrap(),
-            ),
-        )
-        .into()),
-    }
+    })
+}
+
+async fn run_collection<V: vdaf::Collector, Q: QueryTypeExt>(
+    options: Options,
+    vdaf: V,
+    http_client: reqwest::Client,
+    query: Query<Q>,
+    agg_param: &V::AggregationParam,
+) -> Result<(), Error>
+where
+    V::AggregateResult: Debug,
+{
+    let collection = new_collector(options, vdaf, http_client)?
+        .collect(query, agg_param)
+        .await
+        .map_err(|err| Error::Anyhow(err.into()))?;
+    print_collection::<V, Q>(collection)?;
+    Ok(())
 }
 
-async fn run_collection_generic<V: vdaf::Collector, Q: QueryTypeExt>(
+async fn run_new_job<V: vdaf::Collector, Q: QueryTypeExt>(
     options: Options,
     vdaf: V,
     http_client: reqwest::Client,
     query: Query<Q>,
     agg_param: &V::AggregationParam,
+    collection_job_id: CollectionJobId,
+) -> Result<(), Error>
+where
+    V::AggregateResult: Debug,
+{
+    let collection = new_collector(options, vdaf, http_client)?
+        .start_collection_with_id(collection_job_id, query, agg_param)
+        .await
+        .map_err(|err| Error::Anyhow(err.into()))?;
+    println!("Job ID: {}", collection.collection_job_id());
+    Ok(())
+}
+
+async fn run_poll_job<V: vdaf::Collector, Q: QueryTypeExt>(
+    options: Options,
+    vdaf: V,
+    http_client: reqwest::Client,
+    query: Query<Q>,
+    agg_param: V::AggregationParam,
```

Comment:
We may as well standardize the arguments to the different run functions:
```suggestion
    agg_param: &V::AggregationParam,
```

Reply:
I had it this way because of this confusing error:
```
error[E0308]: mismatched types
   --> tools/src/bin/collect.rs:646:20
    |
646 |         .poll_once(&collection_job)
    |          --------- ^^^^^^^^^^^^^^^ expected associated type, found `&<... as Vdaf>::AggregationParam`
    |          |
    |          arguments to this method are incorrect
    |
    = note: expected reference `&CollectionJob<<V as prio::vdaf::Vdaf>::AggregationParam, _>`
               found reference `&CollectionJob<&<V as prio::vdaf::Vdaf>::AggregationParam, Q>`
```

But with a fresh brain I see now I can just `agg_param.clone()` when I construct the `collection_job`.